packages/grpc-js/src/resolver-dns.ts

```diff
@@ -189,7 +194,7 @@ class DnsResolver implements Resolver {
        * because when looking up a single family, dns.lookup outputs an error
        * if the name exists but there are no records for that family, and that
        * error is indistinguishable from other kinds of errors */
-      this.pendingLookupPromise = dnsLookupPromise(hostname, { all: true });
+      this.pendingLookupPromise = this.resolver.resolve(hostname);
```

Comment:
This should match the previous behavior, so it should be `this.resolver.lookup(hostname, { all: true })`. The other changes that this causes should also be reversed.

Reply:
There is not `.lookup` for the Resolver class https://nodejs.org/api/dns.html#class-dnsresolver. I thought it resolves both ipv4 and ipv6 but checking the docs again is defaults to ipv4. We could use `resolveAny` and filter only `A` and `AAA`, wdyth?

Reply:
You're right, I thought I saw it there. Unfortunately, according to the `resolveAny` documentation,

> DNS server operators may choose not to respond to ANY queries. It may be better to call individual methods like [dns.resolve4()](https://nodejs.org/api/dns.html#dnsresolve4hostname-options-callback), [dns.resolveMx()](https://nodejs.org/api/dns.html#dnsresolvemxhostname-callback), and so on.

So we can't use `resolveAny`. As a result, `lookup` is the only function that will consistently get IPv4 and IPv6 results in a single operation.

In addition, as stated in the [Implementation considerations section](https://nodejs.org/api/dns.html#implementation-considerations), `lookup` uses `getaddrinfo` while `resolve` operations directly make DNS queries, probably using c-ares as the C core does. This means that switching between them can introduce other unexpected changes in behavior.

Reply:
hm I see. What would your suggestion then be for this case?

Reply:
I think this is part of the reason the C core has two different DNS configurations, and this behavior is only supported in one of them. One possibility is to mirror that here, with an equivalent to `GRPC_DNS_RESOLVER=ares`.

Reply:
Okay I see I can update the pr with something similar and see how the solution looks like? 

Thinking something (very basic pseudo), so change the behaviour if explicitly set like it. 

```js
if (process.env.GRPC_DNS_TODO_NAME) {
	this.resolver  = dns.ResolverWithDNSServer()
} else {
   this.resolver = dns.lookup()
}
```

Reply:
Have updated the pr where now we can select the behaviour. Also have left the env name as a placeholder.

Reply:
An update on my side with `Any` I don't get any response from our DNS also reading the docs
> DNS server operators may choose not to respond to ANY queries. It may be better to call individual methods like [dns.resolve4()](https://nodejs.org/api/dns.html#dnsresolve4hostname-options-callback), [dns.resolveMx()](https://nodejs.org/api/dns.html#dnsresolvemxhostname-callback), and so on. For more details, see [RFC 8482](https://tools.ietf.org/html/rfc8482).

Would be acceptable calling both methods (`resolve4`, `resolve6`)? Not sure what are the implications there though.

Reply:
@murgatroid99 any thoughts on this? To do something around 

```
merge(await promise(resolve4(), resolve6))
```

Reply:
I'm not sure what you mean by `promise(resolve4(), resolve6()))`. Calling both `resolve4` and `resolve6` should be OK, but I am concerned about one of them returning a result and the other one erroring. So, this should probably use `Promise.allSettled()`.

Reply:
Yes this is what I meant. Cool will work on updating the pr to use this logic and also add some unit tests. 

Also do I understand correct that the new behaviour is going to be used using `GRPC_DNS_RESOLVER=ares` ?

Reply:
I'm not sure about that. Since that setting already exists in the other library, people may have expectations about its behavior, and it's unclear if this implementation will meet that expectations. Maybe it could be something like `GRPC_NODE_USE_ALTERNATIVE_RESOLVER=true|false`

Reply:
Makes sense 👍